src/switch/Switch.tsx

```diff
@@ -0,0 +1,129 @@
+import { cx, useControllableState } from "@renderlesskit/react";
+import {
+  CheckboxStateReturn,
+  Checkbox as RenderlessSwitch,
+  CheckboxProps as RenderlessSwitchProps,
+} from "reakit";
+import * as React from "react";
+
+import { forwardRefWithAs } from "../utils/types";
+import { Box, BoxProps } from "../box";
+import { useTheme } from "../theme";
+
+export type SwitchStatus = CheckboxStateReturn["state"];
+
+export type SwitchStateValues = {
+  state?: SwitchStatus;
+  size?: "xs" | "sm" | "lg";
+};
+
+export type SwitchStateActions = Pick<CheckboxStateReturn, "setState">;
+
+export type SwitchStateProps = SwitchStateValues & SwitchStateActions;
+
+export type SwitchLabelProps = BoxProps & SwitchStateProps;
+
+export const SwitchLabel = forwardRefWithAs<
+  SwitchLabelProps,
+  HTMLLabelElement,
+  "label"
+>((props, ref) => {
+  const { state, setState, size, ...mainProps } = props;
```

Comment:
Any use of state, setState here 🤔

Reply:
I will change these back to context, seems not nice.